core/iwasm/aot/aot_runtime.c

```diff
@@ -1158,6 +1158,11 @@ aot_instantiate(AOTModule *module, bool is_sub_inst, uint32 stack_size,
         (WASMModuleInstanceCommon *)module_inst);
 #endif
 
+#if WASM_ENABLE_WASI_NN != 0
+    ((AOTModuleInstanceExtra *)module_inst->e)->wasi_nn_ctx =
+        wasi_nn_initialize();
+#endif
```

Comment:
Had better move these code before calling `execute_post_inst_function`. How about moving them after L1103, after the initialization of wasi_ctx. Another issue is to check the return value:
```C
#if WASM_ENABLE_LIBC_WASI != 0
    if (!is_sub_inst) {
        if (!wasm_runtime_init_wasi(
....
    }
#endif

#if WASM_ENABLE_WASI_NN != 0
    if (!is_sub_inst) {
        if (!((AOTModuleInstanceExtra *)module_inst->e)->wasi_nn_ctx =
            wasi_nn_initialize())) {
            set_error_buf(error_buf, error_buf_size, "wasi nn initialization failed");
            goto fail;
         }
    }
#endif
```
Adding `if (!is_sub_inst)` is due to multi-threading, if the wasm app of wasi-nn supports multi-thread, each instance's `wasi_nn_ctx` of this wasm app should be the same, the sub instance's `wasi_nn_ctx` should be propagated from main instance's `wasi_nn_ctx`, like the handling of `wasi_ctx`, refer to lib_pthread_wrapper.c:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/libraries/lib-pthread/lib_pthread_wrapper.c#L599-L603

Should add the similar code in lib_pthread_wrapper.c

Reply:
OK, got it